plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,485 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_tenant_igmp_interface_policy
+short_description: Manage Internet Group Management Protocol (IGMP) Interface Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage IGMP Interface Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Anvitha Jain (@anvjain)
+options:
+  template:
+    description:
+    - The name of the tenant template.
+    type: str
+    aliases: [ tenant_template ]
+    required: true
+  name:
+    description:
+    - The name of the IGMP Interface Policy.
+    type: str
+    aliases: [ igmp_interface_policy ]
+  uuid:
+    description:
+    - The UUID of the IGMP Interface Policy.
+    - This parameter is required when the IGMP Interface Policy O(name) needs to be updated.
+    type: str
+  description:
+    description:
+    - The description of the IGMP Interface Policy.
+    - Providing an empty string will remove the O(description="") from the IGMP Interface Policy.
+    type: str
+  version3_asm:
+    description:
+    - Enable or disable IGMP version 3 ASM.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+    aliases: [ allow_version3_asm ]
+  fast_leave:
+    description:
+    - Enable or disable fast leave.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  report_link_local_groups:
+    description:
+    - Enable or disable reporting link-local groups.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  igmp_version:
+    description:
+    - The IGMP version of the IGMP Interface Policy.
+    - Defaults to C(v2) when unset during creation.
+    type: str
+    choices: [ v2, v3 ]
+  group_timeout:
+    description:
+    - The group timeout value in seconds.
+    - Defaults to C(260) when unset during creation.
+    - The valid range is from C(3) to C(65535).
```

Comment:
We did not use `C` for the integer attributes. Are we planning to implement the new change?

Reply:
changed it to just numbers